### PR TITLE
Made canvas generated in canvas instead of png

### PR DESCRIPTION
**Pull request by bestander**
# Description

For https://github.com/carabus/arrow-tracker/issues/7 - Target is now generated from JS rather than using a png file.

# Video

ezgif-1-bf2086669c9b

# Test Plan
- test targeting in desktop browser
- test targeting in iphone
- test targeting in android

### CODE REVIEW

File: client/src/components/target/target.js [hunk 1]
```diff
@@ -1,195 +1,251 @@
 import React from "react";
-import target from "./target.png";
 
 export default class Target extends React.Component {
   constructor(props) {
     super(props);
-    this.setCanvasRef = element => {
-      this.canvas = element;
-    };
+    this.canvasRef = React.createRef();
```
Comment (bestander) [inline client/src/components/target/target.js]: This is a newer way to get ref for an element rendered by React.
this.canvasRef.current accesses the Node and works after rerenders

File: client/src/components/target/target.js [hunk 2]
```diff
@@ -1,195 +1,251 @@
...
   }
 
   render() {
-    return <canvas ref={this.setCanvasRef} width="300" height="300" />;
+    return (
+      <canvas
+        ref={this.canvasRef}
+        width={this.canvasSizePx}
+        height={this.canvasSizePx}
+        onMouseEnter={() => {
```
Comment (bestander) [inline client/src/components/target/target.js]: This is better than previous this.canvas.addEventListerner:
- less code to write
- don't need to removeEventListener on unmounting

File: client/src/components/target/target.js [hunk 3]
```diff
@@ -1,195 +1,251 @@
...
+        }}
+      />
+    );
   }
 
-  centerPoint = { x: 150, y: 150 };
-  touchCursorOffset = 100;
   targetScoreRange = [
     {
-      minRadius: 0,
```
Comment (bestander) [inline client/src/components/target/target.js]: minRadius is unnecessary, we assume that all targets overlap each other

File: client/src/components/target/target.js [hunk 4]
```diff
@@ -1,195 +1,251 @@
...
-    let score = this.targetScoreRange.find(
-      item =>
-        distanceFromCenter <= item.maxRadius &&
-        distanceFromCenter >= item.minRadius
-    );
-
-    return score
-      ? { score: score.points, isInverted: score.invert }
-      : { score: 0, isInverted: false };
+    for (let i = this.targetScoreRange.length - 1; i >= 0; i--) {
```
Comment (bestander) [inline client/src/components/target/target.js]: without minRadius need to traverse the sorted array

**Comment by bestander** [comment]
e2e test  FAIL  src/components/target/target.test.js got broken, will have a look

**Comment by carabus** [comment]
Reviewed, just waiting for mobile testing and for all tests to pass.

**Comment by bestander** [comment]
A bit broken on iOS device, working on a fix

**Comment by bestander** [comment]
Done the final tweaks and tested on mobile phones

**Pull request merged**